src/java_tools/buildjar/java/com/google/devtools/build/java/turbine/update_turbine_pgo.sh

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
+
+# Copyright 2024 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Updates the PGO profile for the Turbine Graal Native Image on a representative
+# target in the Bazel source tree.
+
+set -euo pipefail
+
+REPRESENTATIVE_TARGET="//src/main/java/com/google/devtools/build/lib/skyframe:skyframe_cluster"
+
+cd "$BUILD_WORKSPACE_DIRECTORY"
+
+echo "===== Building turbine with PGO instrumentation ====="
+bazel build \
+  -c opt \
+  --//src/java_tools/buildjar/java/com/google/devtools/build/java/turbine:turbine_direct_graal_pgo_instrument \
+  //src/java_tools/buildjar/java/com/google/devtools/build/java/turbine:turbine_direct_graal
+TURBINE_PATH="$(
+  bazel cquery \
+    --output=files \
+    -c opt \
+    --//src/java_tools/buildjar/java/com/google/devtools/build/java/turbine:turbine_direct_graal_pgo_instrument \
+    //src/java_tools/buildjar/java/com/google/devtools/build/java/turbine:turbine_direct_graal
+)"
+
+echo "===== Collecting profile for $REPRESENTATIVE_TARGET ====="
+
+# The random path serves as a cache breaker for the build below.
+PROFILE="$(pwd)/_turbine_pgo/profile.$(uuid -v4).iprof"
```

Comment:
I don't have a uuid command on my mac or linux machines and it wasn't immediately obvious where to find on, what would you recommend?

Reply:
Maybe uuidgen?

Reply:
Both are available on both my machines, but I switched to `uuidgen` as it seems to be more widespread.